pkg/cmd/roachtest/tests/failover.go

```diff
@@ -70,6 +77,74 @@ func registerFailover(r registry.Registry) {
 	}
 }
 
+// 5 nodes fully connected. Break the connection between a pair of nodes 4 and 5
+// while running a workload against nodes 1 through 3. Before each disconnect,
+// move all the leases to nodes 4 and 5 in a different pattern.
+func runDisconnect(ctx context.Context, t test.Test, c cluster.Cluster) {
+	require.Equal(t, 6, c.Spec().NodeCount)
+	require.False(t, c.IsLocal(), "test can't use local cluster") // messes with iptables
+
+	// Create cluster.
+	opts := option.DefaultStartOpts()
+	settings := install.MakeClusterSettings()
+
+	c.Put(ctx, t.Cockroach(), "./cockroach")
+	c.Start(ctx, t.L(), opts, settings, c.Range(1, 5))
+
+	conn := c.Conn(ctx, t.L(), 1)
+	defer conn.Close()
+
+	// Wait for upreplication.
+	require.NoError(t, WaitFor3XReplication(ctx, t, conn))
+
+	t.Status("creating workload database")
+	_, err := conn.ExecContext(ctx, `CREATE DATABASE kv`)
+	require.NoError(t, err)
+
+	c.Run(ctx, c.Node(6), `./cockroach workload init kv --splits 100 {pgurl:1}`)
+
+	// Start workload on n6 using nodes 1-3 (not part of partition). We could
+	// additionally test the behavior of running SQL against nodes 4-5 however
+	// that complicates the analysis as we want to focus on KV behavior.
+	t.Status("running workload")
+	m := c.NewMonitor(ctx, c.Range(1, 3))
+	m.Go(func(ctx context.Context) error {
+		c.Run(ctx, c.Node(6), `./cockroach workload run kv --read-percent 50 `+
+			`--duration 10m --concurrency 256 --max-rate 2048 --timeout 1m --tolerate-errors `+
+			`--histograms=`+t.PerfArtifactsDir()+`/stats.json `+
+			`{pgurl:1-3}`)
+		return nil
+	})
+
+	// Start and stop partial between nodes 4 and 5 every 30 seconds.
+	m.Go(func(ctx context.Context) error {
+		// Try different combinations of the ranges between the partitioned nodes.
+		// All the system ranges will be on all the nodes, so they will all move,
+		// plus many of the non-system ranges.
+		for i := 2; i <= 7; i++ {
+			time.Sleep(30 * time.Second)
+
+			t.Status("Moving ranges to nodes 4 and 5 before partition", i)
+			relocateLeases(t, ctx, conn, `range_id = 2`, 4)
+			relocateLeases(t, ctx, conn, `voting_replicas @> ARRAY[5] AND range_id != 2`, 5)
+
+			t.Status("disconnecting n4 and n5")
+			Disconnect(t, c, ctx, []int{4, 5}, false)
+
+			time.Sleep(30 * time.Second)
+			qps := measureQPS(ctx, t, conn, 5*time.Second)
+			t.Status("Node 1 QPS after waiting 30 seconds is: ", qps)
+			// There is some rounding error with QPS, so verify that it is > 1
+			require.True(t, qps > 1.0)
```

Comment:
We shouldn't assert anything here. These tests are intended to be more like benchmarks rather than pass/fail tests, since we [graph them with roachperf](https://roachperf.crdb.dev/?filter=&view=failover%2Fnon-system%2Fcrash&tab=gce) and want to track availability improvements over time. Failing the test won't record a data point for that run (I think).